UnityProject/Assets/Scripts/Communications/SignalEmitter.cs

```diff
@@ -4,28 +4,53 @@
 using ScriptableObjects.Communications;
 using UnityEngine;
 using Mirror;
+using NaughtyAttributes;
 
 namespace Communications
 {
 	public abstract class SignalEmitter : NetworkBehaviour
 	{
-		[SerializeField] protected SignalDataSO signalData;
-		[SerializeField] protected float frequency = 122f;
+		[SerializeField]
+		[Required("A signalSO is required for this to work.")]
+		protected SignalDataSO signalData;
+		public EncryptionDataSO EncryptionData;
+		[SerializeField]
+		protected float frequency = 122f;
+		[SerializeField]
+		[Tooltip("For devices that require a power source to operate such as newscasters and wall mounted department radios.")]
+		protected bool requiresPower = false;
+		[SerializeField]
+		[ShowIf(nameof(requiresPower))]
+		protected bool isPowered = true;
```

Comment:
what is this one for?

Reply:
for powered devices, ignore it for now like i said in the notes because I haven't added things like news casters yet.

Reply:
yeah but why two bools

Reply:
Requires power means that the device needs to be connected to a power source to function
is powered means that the device is currently being powered by something and isn't offline

So remote signalers don't require power so they'll work without it
but if we add a device that requires power from an battery or APC to function we can say it requires power then is powered is when the battery isn't empty and the device is still online

Reply:
Good, that's what I thought. Then why is ``IsPowered`` serialized? I t should jsut be the result of ``APCPoweredDevice.HasPower()`` (or whatever it is actually called)

Reply:
Allows mappers and designers to make things powered or unpowered on start if it uses a battery instead of an APC